libsplinter/Cargo.toml

```diff
@@ -161,7 +161,7 @@ biome-credentials = ["bcrypt", "biome"]
 biome-key-management = ["biome"]
 biome-notifications = ["biome"]
 biome-profile = ["biome"]
-challenge-authorization = []
+challenge-authorization = ["trust-authorization"]
```

Comment:
Feels like trust and challenge feature should be at the same level and not dependent upon each other.

Reply:
Updated, removed the commit that made them dependent and properly guarded the rest of the code throughout the commits